pkg/catalog/utils/version_test.go

```diff
@@ -129,3 +132,73 @@ func TestVersionSatifiesRange(t *testing.T) {
 	testInvalidVersion(t, "versionInvalid-1.0", "<=v1.0.0-validVersion")
 
 }
+
+func TestLatestAvailableTemplateVersion(t *testing.T) {
+	template := &v3.CatalogTemplate{
+		Template: v3.Template{
+			Spec: v3.TemplateSpec{
+				Versions: []v3.TemplateVersionSpec{
+					{
+						ExternalID:        "catalog://?catalog=library&template=artifactory-ha&version=0.12.16",
+						Version:           "0.12.16",
+						RancherMinVersion: "v2.2.0",
+						RancherMaxVersion: "v2.3.0",
+					},
+					{
+						ExternalID:        "catalog://?catalog=library&template=artifactory-ha&version=0.12.15",
+						Version:           "0.12.15",
+						RancherMinVersion: "v2.1.0",
```

Comment:
You need to add test for same Rancher range with different version. So that you can make sure your function `LatestAvailableTemplateVersion` will return the latest one.